### ISSUE TITLE
Two level logs

### ISSUE THREAD

**Issue by kamo-naoyuki** [wandb/client#1530]
I'd like to log the values at two levels, I mean,

```python
iiter = 0
for epoch in range(1, 10):
    for batch in data_loader:
        iiter += 1
        loss = model(batch)
        ...
        wandb.log({"loss": loss}, step=iiter)
     ...
     wandb.log({"loss_mean": loss_mean}, step=epoch)
```

However, it shows "wandb: WARNING Step must only increase in log calls".
Is there a way to control two level values?

**Comment by vanpelt** [comment]
Hey @kamo-naoyuki instead of specifying step, you can log the different steps as separate metrics:

```
iiter = 0
for epoch in range(1, 10):
    for batch in data_loader:
        iiter += 1
        loss = model(batch)
        ...
        wandb.log({"loss": loss, "batch": iiter}, commit=False)
     ...
     wandb.log({"loss_mean": loss_mean, "epoch": epoch})
```

Then in the ui you can choose different x-axis to visualize your metrics.

**Comment by kamo-naoyuki** [comment]
Looks good. Thank you. 
I think it's better if some grouping can be done to control different steps.

**Comment by lukas** [comment]
I'm note sure I understand your suggestion - how would you like this to work?

**Comment by kamo-naoyuki** [comment]
For example,

```python
wandb.log({"value1": value1}, step=iteration, group="iteration")
wandb.log({"value2": value2}, step=epoch, group="epoch")
```

Now different categories values are shown in the same level in the UI.

**Comment by mrluin** [comment]
Hello @vanpelt , thanks for you code.
I have tried the case you provided, but the log of inner loop is overwirtten step-by-step that the results on UI only display the last time log. Do you have any suggestions?

**Comment by weizhenFrank** [comment]
@mrluin  Yeah, I have the same problem as yours, is there any good solution?

**Comment by nate-wandb** [comment]
@mrluin @weizhenFrank 
Are you logging with the same metric names? In Van Pelt's solution he is calling the metrics `loss` and `loss_mean` so that they do not overwrite each other. 

Both the metrics logged this way should still have a `history` where you can see the metric at a particular step but for `loss` a step would be one batch and for `loss_mean` a step would be one epoch.

Let me know if that helps clarify or if I'm not answering your question.

**Comment by sydholl** [comment]
We are closing this issue due to inactivity, please comment to reopen.
